astropy/table/table.py

```diff
@@ -2211,3 +2219,45 @@ def _convert_col_for_table(self, col):
             col = super(QTable, self)._convert_col_for_table(col)
 
         return col
+
+class NdarrayMixin(np.ndarray):
+    """
+    Minimal mixin using a simple subclass of numpy array
+    """
+    info = InfoDescriptor(ParentDtypeInfo)
+
+    def __new__(cls, obj, *args, **kwargs):
+        self = np.array(obj, *args, **kwargs).view(cls)
+        if 'info' in getattr(obj, '__dict__', ()):
+            self.info = obj.info
+        return self
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+
+        if six.callable(super(NdarrayMixin, self).__array_finalize__):
+            super(NdarrayMixin, self).__array_finalize__(obj)
+
+        # Self was created from template (e.g. obj[slice] or (obj * 2))
+        # or viewcast e.g. obj.view(Column).  In either case we want to
+        # init Column attributes for self from obj if possible.
+        if 'info' in getattr(obj, '__dict__', ()):
+            self.info = obj.info
+
+    def __reduce__(self):
```

Comment:
Just a wild guess (though the link below might explain it better), it seems to me the ndarray `__setstate__` probably falls over if we pass in additional state variables that it doesn't know about.  So we have to separate the `ndarray` specific state variables from `ndarray.__reduce__` from the ones specific to our subclass, and then reconstitute them separately in `__setstate__`.